src/main/java/frc/robot/superstructure/Superstructure.java

```diff
@@ -46,62 +56,149 @@ public Command setState(RobotState state) {
 	//@formatter:off
 	public Command idle() {
 		return new ParallelCommandGroup(
-				swerve.getCommandsBuilder().saveState(SwerveState.DEFAULT_DRIVE)
+				swerve.getCommandsBuilder().saveState(SwerveState.DEFAULT_DRIVE),
+				funnelStateHandler.setState(FunnelState.STOP)
+				//elevatorRoller.STOP
+				//intakeRoller.STOP
+				//flywheel.DEFAULT
+				//intakePivot.IDLE
+				//elevator.IDLE
 		);
 	}
 
 	public Command intake() {
 		return new ParallelCommandGroup(
+				new SequentialCommandGroup(
+						new ParallelCommandGroup(
+								//intakePivot.INTAKE
+								//intakeRoller.INTAKE
+								funnelStateHandler.setState(FunnelState.STOP)
+						).withTimeout(3), // .until(() -> isNoteInIntake
+						new ParallelCommandGroup(
+								funnelStateHandler.setState(FunnelState.NOTE_TO_SHOOTER)
+								//intakePivot.IDLE
+						).until(this::isNoteInShooter),
+						new ParallelCommandGroup(
+								//intakeRoller.STOP
+								funnelStateHandler.setState(FunnelState.STOP)
+						)
+				),
 				swerve.getCommandsBuilder().saveState(SwerveState.DEFAULT_DRIVE.withAimAssist(AimAssist.NOTE))
+				//flywheel.DEFAULT
+				//elevator.IDLE
+				//elevatorRoller.STOP
 		);
 	}
 
 	public Command preSpeaker() {
 		return new ParallelCommandGroup(
-				swerve.getCommandsBuilder().saveState(SwerveState.DEFAULT_DRIVE.withAimAssist(AimAssist.SPEAKER))
+				swerve.getCommandsBuilder().saveState(SwerveState.DEFAULT_DRIVE.withAimAssist(AimAssist.SPEAKER)),
+				//flywheel.SHOOT
+				funnelStateHandler.setState(FunnelState.STOP)
+				//intakeRoller.STOP
+				//elevator.IDLE
+				//intakePivot.IDLE
+				//elevatorRoller.STOP
 		);
 	}
 
 	public Command speaker() {
 		return new ParallelCommandGroup(
+				new SequentialCommandGroup(
+						funnelStateHandler.setState(FunnelState.STOP).withTimeout(3), // .until(() -> isReadyToShoot())
+						funnelStateHandler.setState(FunnelState.SPEAKER).until(() -> !isNoteInShooter()),
+						funnelStateHandler.setState(FunnelState.STOP)
+				),
+				//flywheel.SHOOT
+				//intakeRoller.STOP
+				//elevator.IDLE
+				//intakePivot.IDLE
+				//elevatorRoller.STOP
 				swerve.getCommandsBuilder().saveState(SwerveState.DEFAULT_DRIVE.withAimAssist(AimAssist.SPEAKER))
 		);
 	}
 
 	public Command preAmp() {
 		return new ParallelCommandGroup(
-				swerve.getCommandsBuilder().saveState(SwerveState.DEFAULT_DRIVE.withAimAssist(AimAssist.AMP))
+				new ParallelCommandGroup(
+						swerve.getCommandsBuilder().saveState(SwerveState.DEFAULT_DRIVE.withAimAssist(AimAssist.AMP)),
+						funnelStateHandler.setState(FunnelState.STOP)
+						//elevator.PRE_SCORE
+				).withTimeout(3) // .until(() -> isReadyToAmp())
```

Comment:
Make timeouts class like tolerances class

Reply:
Why? The timeouts are only temporary, once a certain subsystem will be ready we will just replace it with the currect function